test/utils/asserts.hpp

```diff
@@ -11,6 +11,14 @@
         EXPECT_EQ(r.stack[0], (expected)); \
     } while (false)
 
+#define EXPECT_EMPTY_RESULT(result)   \
+    do                                \
+    {                                 \
+        const auto r = (result);      \
+        ASSERT_FALSE(r.trapped);      \
```

Comment:
It's called `EXPECT_` but in fact does `ASSERT`.
I think I would change both `ASSERT`s to `EXPECT`s

Reply:
Yeah assert vs expect is plagued all over the test suite. if we want to be fully consistent then trapped should still be assert, but the size can be changed to expect. (See `EXPECT_RESULT`).